### PR TITLE
Extracts compatibility-flags out of compatibility-dates

### PR DESCRIPTION
**Pull request by vicb**
### Summary

While the concepts are related they are not the same.

compatibility-flags will grow as we had more flags and deserves a separate page. This also has the benefit to have add an entry in the side bar. Having to click on "Configuration > Compatibility dates" to get info on the Compatibility flags is counter-intuitive.

I have updated internal links to point to the newly added flag page. Unfortunately external links to a given flag (i.e. "workers/configuration/compatibility-dates/#nodejs-compatibility-flag") will not point to that flag but to the top of the "Compatibility date". This is a limitation of the redirect mechanism on the site which can not redirect a fragment.

There are only a few changes in the content for the new flag page. I extracted a sentence from the first section into as the subtitle and I fixed a typo. The only other change is updating the CODEOWNERS of the new page to the same as the page is was extracted from.

### Documentation checklist

- [x] The [documentation style guide](https://developers.cloudflare.com/style-guide/) has been adhered to.
- [x] Files which have changed name or location have been allocated [redirects](https://developers.cloudflare.com/pages/configuration/redirects/#per-file) - see the commit message for more info.

### CODE REVIEW

File: src/content/docs/workers/configuration/compatibility-dates.mdx [hunk 1]
```diff
@@ -43,68 +43,3 @@ The compatibility date cannot be updated on the Cloudflare Dashboard at this tim
 The compatibility date can be set when uploading a Worker using the [Workers Script API](/api/operations/worker-script-upload-worker-module) or [Workers Versions API](/api/operations/worker-versions-upload-version#request-body) in the request body's `metadata` field.
 
 If a compatibility date is not specified on upload via the API, it defaults to the oldest compatibility date, before any flags took effect (2021-11-02). When creating new Workers, it is highly recommended to set the compatibility date to the current date when uploading via the API.
-
```
Comment (irvinebroque) [inline src/content/docs/workers/configuration/compatibility-dates.mdx:46]: This page needs to have a clearer reference (and link) to compatibility flags if we are removing that content from here and putting it on another page

File: src/content/docs/workers/configuration/compatibility-flags.mdx [hunk 2]
```diff
@@ -0,0 +1,72 @@
+---
+pcx_content_type: concept
+title: Compatibility flags
+rss: https://github.com/cloudflare/cloudflare-docs/commits/production/src/content/compatibility-flags.atom
+head: []
+description: Opt into a specific features of the Workers runtime for your Workers project.
+---
+
+import { CompatibilityDates } from "~/components";
+
```
Comment (irvinebroque) [inline src/content/docs/workers/configuration/compatibility-flags.mdx:10]: Somewhere on this page, need to link to compatibility dates and explain the connection between the two.

File: src/content/docs/workers/testing/vitest-integration/isolation-and-concurrency.mdx [hunk 3]
```diff
@@ -56,7 +56,7 @@ In this model, a single `workerd` process is started with a single Worker for al
...
+The Workers Vitest pool works by running code inside a Cloudflare Worker that Vitest would usually run inside a [Node.js worker thread](https://nodejs.org/api/worker_threads.html). To make this possible, the pool requires the [`nodejs_compat`](/workers/configuration/compatibility-flags/#nodejs-compatibility-flag) and [`export_commonjs_default`](/workers/configuration/compatibility-flags/#commonjs-modules-do-not-export-a-module-namespace) compatibility flags to be enabled. The pool also configures `workerd` to use Node-style module resolution and polyfills required `node:*` modules not provided by `nodejs_compat`.
```
[the remark this answers is not in the document]
Comment (vicb) [reply src/content/docs/workers/testing/vitest-integration/isolation-and-concurrency.mdx:59]: This is existing content

File: src/content/docs/workers/testing/vitest-integration/configuration.mdx [hunk 4]
```diff
@@ -145,7 +145,7 @@ The following functions are exported from the `@cloudflare/vitest-pool-workers/c
...
 
-  * Use this to provide configuration information that is typically stored within the Wrangler configuration file, such as [bindings](/workers/runtime-apis/bindings/), [compatibility dates](/workers/configuration/compatibility-dates/), and [compatibility flags](/workers/configuration/compatibility-dates/#compatibility-flags). The `WorkerOptions` interface is defined [here](https://github.com/cloudflare/workers-sdk/tree/main/packages/miniflare#interface-workeroptions). Use the `main` option above to configure the entry point, instead of the Miniflare `script`, `scriptPath`, or `modules` options.
+  * Use this to provide configuration information that is typically stored within the Wrangler configuration file, such as [bindings](/workers/runtime-apis/bindings/), [compatibility dates](/workers/configuration/compatibility-dates/), and [compatibility flags](/workers/configuration/compatibility-flags/). The `WorkerOptions` interface is defined [here](https://github.com/cloudflare/workers-sdk/tree/main/packages/miniflare#interface-workeroptions). Use the `main` option above to configure the entry point, instead of the Miniflare `script`, `scriptPath`, or `modules` options.
```
[the remark this answers is not in the document]
Comment (vicb) [reply src/content/docs/workers/testing/vitest-integration/configuration.mdx:148]: This is existing content

File: src/content/docs/workers/configuration/compatibility-dates.mdx (hunk 1 above)
Comment (vicb) [reply src/content/docs/workers/configuration/compatibility-dates.mdx:46]: I think the reference was already there (on line 13 of the current version).

But adding a link is a good point, I'll push a fixup commit to this PR.

File: src/content/docs/workers/configuration/compatibility-flags.mdx (hunk 2 above)
Comment (vicb) [reply src/content/docs/workers/configuration/compatibility-flags.mdx:10]: I'll reuse the same sentence than on the date page (l 13 referenced above) here and link the date.

**Comment by vicb** [comment]
@irvinebroque Thanks for the review, I have updated the content, please take another look.

**Comment by vicb** [comment]
_PR rebased to resolve a merge conflict_

**Review by irvinebroque — APPROVED** [review]

**Comment by vicb** [comment]
PR rebased to resolve a merge conflict

**Comment by irvinebroque** [comment]
@ToriLindsay @dcpena - can we get ✅ to merge this?

**Review by dcpena — APPROVED** [review]

**Comment by vicb** [comment]
Thanks all for the help getting this merged

**Pull request merged**